hash_version.sh

```diff
@@ -1,8 +1,10 @@
-#!/bin/bash
+#!/bin/sh
 
-GIT_HASH=`sh ./calc_hash.sh`;
+GIT_HASH=$(sh ./calc_hash.sh)
 
+# NOTICE:
+# don't use sed -i, because it's not portable between Linux, OS X and some BSDs
 rm -f ver
-sed -b "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdoc/data/resource.h > ver && mv ver renderdoc/data/resource.h
-sed -b "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdocui/Properties/AssemblyInfo.cs > ver && mv ver renderdocui/Properties/AssemblyInfo.cs
+sed "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdoc/data/resource.h > ver && mv ver renderdoc/data/resource.h
```

Comment:
-b gives error on OS X "sed: illegal option -- b", on Linux it's silently
ignored
I've checked manpages on few Linux distros, OS X and BSD and there was no
info about "-b".

As for -i: you are not editing files in-place but moving them, so someone
later could be tempted to make it just use sed -i instead those "mv", but
sed -i is sadly not portable (on some patforms you have to use sed -i"" and
one some sed -i "" - note the space).


2016-08-09 19:22 GMT+02:00 Baldur Karlsson <notifications@github.com>:

> In hash_version.sh
> <https://github.com/baldurk/renderdoc/pull/318#discussion_r74103492>:
>
> >  rm -f ver
> > -sed -b "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdoc/data/resource.h > ver && mv ver renderdoc/data/resource.h
> > -sed -b "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdocui/Properties/AssemblyInfo.cs > ver && mv ver renderdocui/Properties/AssemblyInfo.cs
> > +sed "s/NO_GIT_COMMIT_HASH_DEFINED/$GIT_HASH$1/" renderdoc/data/resource.h > ver && mv ver renderdoc/data/resource.h
>
> Without -b this will change CRLF to LF on windows. Does it not work on
> other platforms? The comment mentions -i which wasn't being used I don't
> think.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/baldurk/renderdoc/pull/318/files/6010982bd4aab9561477f3b6af44c253ba33d32a#r74103492>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA6jxV1uM8Fb7wdGTD3gfAxc6aexa1YQks5qeLdrgaJpZM4Jf9en>
> .
>